x/ccv/provider/keeper/double_vote.go

```diff
@@ -0,0 +1,117 @@
+package keeper
+
+import (
+	"bytes"
+	"fmt"
+
+	errorsmod "cosmossdk.io/errors"
+	cryptotypes "github.com/cosmos/cosmos-sdk/crypto/types"
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	sdkerrors "github.com/cosmos/cosmos-sdk/types/errors"
+	"github.com/cosmos/interchain-security/v2/x/ccv/provider/types"
+	ccvtypes "github.com/cosmos/interchain-security/v2/x/ccv/types"
+	tmtypes "github.com/tendermint/tendermint/types"
+)
+
+// HandleConsumerDoubleVoting verifies a double voting evidence for a given a consumer chain ID
+// and a public key and, if successful, executes the jailing of the malicious validator.
+func (k Keeper) HandleConsumerDoubleVoting(
+	ctx sdk.Context,
+	evidence *tmtypes.DuplicateVoteEvidence,
+	chainID string,
+	pubkey cryptotypes.PubKey,
+) error {
+	// verifies the double voting evidence using the consumer chain public key
+	if err := k.VerifyDoubleVotingEvidence(ctx, *evidence, chainID, pubkey); err != nil {
+		return err
+	}
+
+	// get the validator's consensus address on the provider
+	providerAddr := k.GetProviderAddrFromConsumerAddr(
```

Comment:
Is verification of signatures computationally expensive? If yes, we might want to first verify that the validator address exists and then call `VerifyDoubleVotingEvidence`?